packages/js/src/utils/create-async-iterable/create-async-iterable.spec.ts

```diff
@@ -0,0 +1,57 @@
+import { createAsyncIterable } from './create-async-iteratable';
+
+describe(createAsyncIterable.name, () => {
+  test('simple callback', async () => {
+    const it = createAsyncIterable<string>(({ next, done }) => {
+      setTimeout(() => next('bye'), 10);
+      setTimeout(() => next('hello'), 0);
+      setTimeout(() => done(), 20);
+      setTimeout(() => next('ignored'), 30);
+    });
+    const results: string[] = [];
+    for await (const x of it) {
+      results.push(x);
+    }
+
+    expect(results).toEqual(['hello', 'bye']);
+  });
+
+  test('throwing error', async () => {
+    const it = createAsyncIterable(({ next, error }) => {
+      setTimeout(() => next('hello'), 0);
+      setTimeout(() => error(new Error('Oops')), 10);
+    });
+
+    await expect(async () => {
+      for await (const _x of it) {
+        // nothing
+      }
+    }).rejects.toThrow(/Oops/);
+  });
+
+  test('forked process', async () => {
```

Comment:
I don't think this test is needed. There is nothing in the iterable that is process-specific, right?

Reply:
No, this was just a POC, isn't neccesary.